File: examples/Fortran/OpenMP_basic_offloading.f90
program basic_offloading
    use iso_fortran_env, only : real64
    implicit none    
    
    integer                                      :: i, sys_size
    real(kind=real64), allocatable, dimension(:) :: array

    sys_size = 100000000

    allocate(array(sys_size))

    !$omp target teams distribute parallel do simd
    do i=1, sys_size
        array(i) = real(i)
    enddo
    
    print *, "array(42) = ", array(42)
end program basic_offloading

File: examples/C/OpenMP_basic_offloading.c
#include <stdio.h>
#include <stdlib.h>

int main(void)
{
    int size = 100000000;
    double* array = (double*) malloc(size*sizeof(double));

// We need to explicitly manage transfers in C with NVIDIA compilers
// otherwise we get a runtime error (as of nvhpc 21.9)
#pragma omp target teams distribute parallel for simd map(from:array[0:size])
    for (int i=0; i<size; ++i)
        array[i] = (double) i;

    printf("array[42] = %f\n", array[42]);
    free(array);
    return 0;
}
